torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -57,10 +79,15 @@ void DistAutogradContainer::releaseContext(int64_t context_id) {
       "Could not find autograd context with id: ",
       context_id);
   autograd_context_.erase(context_id);
+
+  if (current_context_id_ == context_id) {
+    // Reset the thread_local current context id, since it is no longer valid.
+    current_context_id_ = -1;
```

Comment:
Will this work for reentrant backward without nested context? (of course, reentrant backward support does not need to be part of the first release)

Reply:
The user will probably invoke a specialized `torch.distributed.backward()` API. This will under the covers only have one autograd context. The engine itself can run multiple re-entrant backward passes under the hood, but we will associate all those backward passes with a single context.